### PR TITLE
Fixes #1987, Improve the error handling when site extension installs fail via ARM

### LINKED ISSUE #1987: Improve the error handling when site extension installs fail via ARM
In [this code](https://github.com/projectkudu/kudu/blob/f155220791f19d7d00d24afe7ba2caa3dae18bc3/Kudu.Core/SiteExtensions/SiteExtensionManager.cs#L450), we send back a very generaic message with no info about the actual exception, amkingmaking it hard to diagnose.

### PR DESCRIPTION
**Pull request by shrimpy**
````
{
  "id": "/subscriptions/fb2c25dc-6bab-45c4-8cc9-cece7c42a95a/resourceGroups/Default-Web-EastUS/providers/Microsoft.Web/sites/kudutry/siteextensions/filecounter",
  "name": "kudutry/filecounter",
  "type": "Microsoft.Web/sites/siteextensions",
  "location": "East US",
  "properties": {
    "id": "filecounter",
    "title": null,
    "type": "Gallery",
    "summary": null,
    "description": null,
    "version": null,
    "extension_url": null,
    "project_url": null,
    "icon_url": null,
    "license_url": null,
    "feed_url": null,
    "authors": null,
    "published_date_time": null,
    "download_count": 0,
    "local_is_latest_version": null,
    "local_path": null,
    "installed_date_time": null,
    "provisioningState": "Failed",
    "comment": "System.IO.FileNotFoundException: Test FileNotFoundException\r\n   at Kudu.Core.SiteExtensions.SiteExtensionManager.<TryInstallExtension>d__27.MoveNext()"
  }
}
````

````
{
  "id": "/subscriptions/fb2c25dc-6bab-45c4-8cc9-cece7c42a95a/resourceGroups/Default-Web-EastUS/providers/Microsoft.Web/sites/kudutry/siteextensions/filecounter",
  "name": "kudutry/filecounter",
  "type": "Microsoft.Web/sites/siteextensions",
  "location": "East US",
  "properties": {
    "id": "filecounter",
    "title": null,
    "type": "Gallery",
    "summary": null,
    "description": null,
    "version": null,
    "extension_url": null,
    "project_url": null,
    "icon_url": null,
    "license_url": null,
    "feed_url": null,
    "authors": null,
    "published_date_time": null,
    "download_count": 0,
    "local_is_latest_version": null,
    "local_path": null,
    "installed_date_time": null,
    "provisioningState": "Failed",
    "comment": "System.Net.WebException: Test WebException\r\n   at Kudu.Core.SiteExtensions.SiteExtensionManager.<TryInstallExtension>d__27.MoveNext()"
  }
}
````

````
{
  "id": "/subscriptions/fb2c25dc-6bab-45c4-8cc9-cece7c42a95a/resourceGroups/Default-Web-EastUS/providers/Microsoft.Web/sites/kudutry/siteextensions/filecounter",
  "name": "kudutry/filecounter",
  "type": "Microsoft.Web/sites/siteextensions",
  "location": "East US",
  "properties": {
    "id": "filecounter",
    "title": null,
    "type": "Gallery",
    "summary": null,
    "description": null,
    "version": null,
    "extension_url": null,
    "project_url": null,
    "icon_url": null,
    "license_url": null,
    "feed_url": null,
    "authors": null,
    "published_date_time": null,
    "download_count": 0,
    "local_is_latest_version": null,
    "local_path": null,
    "installed_date_time": null,
    "provisioningState": "Failed",
    "comment": "Kudu.Contracts.SiteExtensions.InvalidEndpointException: Test InvalidEndpointException\r\n   at Kudu.Core.SiteExtensions.SiteExtensionManager.<TryInstallExtension>d__27.MoveNext()"
  }
}

````

````
{
  "id": "/subscriptions/fb2c25dc-6bab-45c4-8cc9-cece7c42a95a/resourceGroups/Default-Web-EastUS/providers/Microsoft.Web/sites/kudutry/siteextensions/filecounter",
  "name": "kudutry/filecounter",
  "type": "Microsoft.Web/sites/siteextensions",
  "location": "East US",
  "properties": {
    "id": "filecounter",
    "title": null,
    "type": "Gallery",
    "summary": null,
    "description": null,
    "version": null,
    "extension_url": null,
    "project_url": null,
    "icon_url": null,
    "license_url": null,
    "feed_url": null,
    "authors": null,
    "published_date_time": null,
    "download_count": 0,
    "local_is_latest_version": null,
    "local_path": null,
    "installed_date_time": null,
    "provisioningState": "Failed",
    "comment": "System.Exception: Test Exception\r\n   at Kudu.Core.SiteExtensions.SiteExtensionManager.<TryInstallExtension>d__27.MoveNext()"
  }
}
````

### CODE REVIEW

File: Common/Constants.cs [hunk 1]
```diff
@@ -84,8 +84,6 @@ public static TimeSpan MaxAllowedExecutionTime
 
         // TODO: need localization?
         public const string SiteExtensionProvisioningStateNotFoundMessageFormat = "'{0}' not found.";
```
Comment (shrimpy) [inline Common/Constants.cs]: SiteExtensionProvisioningStateNotFoundMessageFormat  is still using in [SiteExtensionManager.cs](https://github.com/projectkudu/kudu/pull/1992/files#diff-afacb573b03286d1f84a90cda5b57ddaR468)

**Comment by davidebbo** [comment]
Looks good. Have you tried trigger an error with the new code to get a sense of what the error response will look like? If needed, just use a hacked Kudu build that throws in some random place just to trigger the exception handling.

**Comment by shrimpy** [comment]
@davidebbo  yes, just updated the PR to include sample output from my verification. what I did was hack kudu to throw error like this ```` throw new Exception("Test Exception") ````

**Comment by davidebbo** [comment]
Looks good!

One minor issue (not really related): would be nice to populate the `feed_url` in the `SiteExtensionInfo` object you create. When I first looked at customer issue, I though they were not setting it because it was null, when in fact they were.

**Comment by shrimpy** [comment]
null feed_url input is a valid case, in this case Kudu will [default to siteextensions.net](https://github.com/projectkudu/kudu/blob/f155220791f19d7d00d24afe7ba2caa3dae18bc3/Kudu.Core/SiteExtensions/SiteExtensionManager.cs#L346)

if we return feed_url, I think we should return the actual feed_url that kudu use. what do you think?

**Comment by CamyDee** [comment]
:smile:

**Pull request merged**